blog-posts/github-billing-dashboard.md

```diff
@@ -0,0 +1,169 @@
+    tick={{ fill: "white" }}
+    axisLine={false}
+    tickLine={false}
+    minTickGap={10}
+  />
+  <YAxis
+    domain={[0, maxValueOfYAxis]}
+    unit=" $"
+    tickCount={maxValueOfYAxis + 1}
+    tick={{ fill: "white" }}
+    axisLine={false}
+    tickLine={false}
+  />
+</BarChart>
+```
+
+### date-fns
+
+For grouping costs by days, weeks and months we had to work a lot with dates. The library `date-fns` made this work easier for us because it brings important functions like `startOfMonth()`, `getISOWeek()`, or `lightFormat()`, which allows you to customize the date format easily. For example, we used the `lightFormat()` function to display the start or end date of the data set. Due to the detailed documentation, it was very pleasant to work with `date-fns`.
+
+```tsx
+isFirstMonth()
+	? ` (from ${lightFormat(new Date(firstDayOfMonth), "dd.MM.")})`
+	: ""
+isLastMonth()
+	? ` (till ${lightFormat(new Date(lastDayOfMonth), "dd.MM.")})`
+	: ""
+```
+
+### react-medium-image-zoom
+
+React-medium-image-zoom is a small but very useful library to zoom in images that are clicked. This library also leaves many possibilities for customization, for example, the background of the zoomed image and the zoom effect can be customized very easily. For the basic function of react-medium-image-zoom it is sufficient to just wrap an image or div tag with the Zoom tag as shown in the example below. The customization features can be added via props.
+
+```tsx
+<Zoom>
+	<img src="sample.jpg"/>
+</Zoom>
+```
+
+### GitHub Pages deployment
+
+We chose GitHub Pages to host the page, as it's a simple and free hosting solution. It was important to us that new features are deployed automatically to GitHub Pages. To automate this process, we decided to use the Github Actions with [GitHub Pages Deploy Action](https://github.com/JamesIves/github-pages-deploy-action).
+This ensures that with every commit to a selected branch (in our case to `main`), the new version is automatically deployed to the `gh-pages` branch. Thereby all changes are immediately visible to the user and can be used directly in the finished product. By using this plugin we also didn't need to change any scripts or code we didn't already have.
+
+So, if you have a React SPA you can just copy&past the GitHub Action and start deploying to GitHub pages. You only need to change the `PUBLIC_URL` variable to your GitHub Pages deployment URL:
+
+```yaml
+name: Deploy to gh-pages
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
+
+      - name: Install and Build 🔧
+        run: |
+          npm install
+          npm run build
+        env:
+          PUBLIC_URL: /github-billing-dashboard
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@4.1.3
+        with:
+          branch: gh-pages
+          folder: build
+```
```

Comment:
This is the end ? Please give me a conclusion 🌟

Reply:
Yes definitely true. We have now added a short summary

Reply:
And I like it. Thanks!